services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1174,6 +1182,79 @@ private Integer fetchSystemStoreSchemaId(String clusterName, String storeName, S
     return response.getId();
   }
 
+  static boolean isPushJobFailedDueToUserError(PushJobDetailsStatus status, PushJobDetails pushJobDetails) {
+    if (PushJobDetailsStatus.isFailed(status)) {
+      PushJobCheckpoints checkpoint = PushJobCheckpoints.valueOf(pushJobDetails.getPushJobLatestCheckpoint());
+      if (checkpoint != null) {
+        Map<CharSequence, CharSequence> pushJobConfigs = pushJobDetails.getPushJobConfigs();
+        Utf8 userErrorCheckpointsKey = new Utf8("push.job.failure.checkpoints.to.define.user.error");
+        Set<PushJobCheckpoints> userErrorCheckpoints = new HashSet<>();
+        if (pushJobConfigs.containsKey(userErrorCheckpointsKey)) {
+          String userErrorCheckpointsStr = pushJobConfigs.get(userErrorCheckpointsKey).toString();
+          // extract individual checkpoints from the string
+          for (String checkpointStr: userErrorCheckpointsStr.split(",")) {
+            checkpointStr = checkpointStr.trim();
+            try {
+              userErrorCheckpoints.add(PushJobCheckpoints.valueOf(checkpointStr));
+            } catch (Exception e) {
+              LOGGER.error(
+                  "Invalid checkpoint {} in the input to define user error. Using the default error checkpoints",
+                  checkpointStr);
+              userErrorCheckpoints.clear();
+              break;
+            }
+          }
+        }
+        if (!userErrorCheckpoints.isEmpty()) {
+          return userErrorCheckpoints.contains(checkpoint);
+        } else {
+          return PushJobCheckpoints.isUserError(checkpoint);
+        }
+      }
+    }
+    return false;
+  }
```

Comment:
This function is overly complicated. Let's extract the list of user configs in `VeniceControllerClusterConfig` (with default value being the one in the enum) and here we should just check if the provided checkpoint is from that list